api/models/workshop.py

```diff
@@ -23,6 +25,10 @@ class Workshop(CapacityMixin, Base):
         WorkshopDifficulty,
         verbose_name=_("Difficulty"),
     )
+    location = models.ForeignKey(
```

Comment:
Why? Workshop locations usually doesn't change between years. The gym is the same gym the next year. The workshop could have same name, but the content, lector and participants can change significantly. Or those thinks can change only slightly every year, but at the end it is completely different workshop.

Maybe I am just missing something substantial about how you want to use it. But you should explain yourself a bit more than in one sentence.